scipy/linalg/decomp_qr.py

```diff
@@ -393,24 +387,15 @@ def rq(a, overwrite_a=False, lwork=None, mode='full', check_finite=True):
     gor_un_grq, = get_lapack_funcs(('orgrq',), (rq,))
 
     if N < M:
-        # get optimal work array
-        Q, work, info = gor_un_grq(rq[-N:], tau, lwork=-1, overwrite_a=1)
-        lwork = work[0].real.astype(numpy.int)
-        Q, work, info = gor_un_grq(rq[-N:], tau, lwork=lwork, overwrite_a=1)
+        Q, = safecall(gor_un_grq, "gorgrq/gungrq", rq[-N:], tau, lwork=lwork,
```

Comment:
This won't work, ``lwork`` will be the one determined for ``gerqf`` above. This call and the ones below should have explicit ``lwork=-1``. 

Seems like it also needs to be documented that ``lwork`` is for lapack function ``gerqf``.